packages/pages/src/vite-plugin/modules/plugin.ts

```diff
@@ -0,0 +1,173 @@
+import { build, createLogger } from "vite";
+import { ProjectStructure } from "../../common/src/project/structure.js";
+import { glob } from "glob";
+import path from "node:path";
+import fs from "node:fs";
+import { convertToPosixPath } from "../../common/src/template/paths.js";
+import { processEnvVariables } from "../../util/processEnvVariables.js";
+import { nodePolyfills } from "vite-plugin-node-polyfills";
+import pc from "picocolors";
+import { addResponseHeadersToConfigYaml } from "../../util/editConfigYaml.js";
+import SourceFileParser, {
+  createTsMorphProject,
+} from "../../common/src/parsers/sourceFileParser.js";
+
+const wrappedCode = (moduleName: string, containerName: string): string => {
+  return `\nconst moduleContainerForBuildUseOnly = document.getElementById('${containerName}');
+if (!moduleContainerForBuildUseOnly) {
+  throw new Error('could not find ${containerName} element');
+}
+
+ReactDOM.render(
+  <React.StrictMode>
+    <${moduleName}/>
+  </React.StrictMode>,
+  moduleContainerForBuildUseOnly
+);
+  `;
+};
+
+const moduleResponseHeader = {
+  pathPattern: "^modules/.*",
+  headerKey: "Access-Control-Allow-Origin",
+  headerValues: ["*"],
+};
+
+export const buildModules = async (
+  projectStructure: ProjectStructure
+): Promise<void> => {
+  if (!shouldBundleModules(projectStructure)) {
+    return;
+  }
+
+  addResponseHeadersToConfigYaml(projectStructure, moduleResponseHeader);
+
+  const { rootFolders, subfolders, envVarConfig } = projectStructure.config;
+  const outdir = path.join(rootFolders.dist, subfolders.modules);
+
+  const filepaths: { [s: string]: string } = {};
+  glob
+    .sync(
+      convertToPosixPath(
+        path.join(rootFolders.source, subfolders.modules, "**/*.{jsx,tsx}")
+      ),
+      { nodir: true }
+    )
+    .forEach((f) => {
+      const filepath = path.resolve(f);
+      const moduleName = getModuleName(filepath);
+      filepaths[moduleName] = filepath;
+    });
+
+  const logger = createLogger();
+  const loggerInfo = logger.info;
+
+  for (const [moduleName, modulePath] of Object.entries(filepaths)) {
+    const index = addExtraModuleCode(modulePath, moduleName);
+    logger.info = (msg, options) => {
+      if (msg.includes("building for production")) {
+        loggerInfo(pc.green(`\nBuilding ${moduleName} module...`));
+        return;
+      }
+      loggerInfo(msg, options);
+    };
+
+    await build({
+      customLogger: logger,
+      configFile: false,
+      envDir: envVarConfig.envVarDir,
+      envPrefix: envVarConfig.envVarPrefix,
+      resolve: {
+        conditions: ["worker", "webworker"],
+      },
+      publicDir: false,
+      css: {
+        postcss: path.join(
+          rootFolders.source,
+          subfolders.modules,
+          `${moduleName}/postcss.config.cjs`
```

Comment:
A user could technically use .cjs or .mjs. or .js. or .ts.

Reply:
Ok! Do we want to support all of them or just .js?

Reply:
Added support for all!